// @azure/arm-rediscache/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
export var acceptLanguage = {
    parameterPath: "acceptLanguage",
    mapper: {
        serializedName: "accept-language",
        defaultValue: 'en-US',
        type: {
            name: "String"
        }
    }
};
export var apiVersion = {
    parameterPath: "apiVersion",
    mapper: {
        required: true,
        serializedName: "api-version",
        type: {
            name: "String"
        }
    }
};
export var cacheName = {
    parameterPath: "cacheName",
    mapper: {
        required: true,
        serializedName: "cacheName",
        type: {
            name: "String"
        }
    }
};
export var defaultParameter = {
    parameterPath: "defaultParameter",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "default",
        defaultValue: 'default',
        type: {
            name: "String"
        }
    }
};
export var history = {
    parameterPath: "history",
    mapper: {
        required: true,
        serializedName: "history",
        type: {
            name: "Number"
        }
    }
};
export var linkedServerName = {
    parameterPath: "linkedServerName",
    mapper: {
        required: true,
        serializedName: "linkedServerName",
        type: {
            name: "String"
        }
    }
};
export var name = {
    parameterPath: "name",
    mapper: {
        required: true,
        serializedName: "name",
        type: {
            name: "String"
        }
    }
};
export var nextPageLink = {
    parameterPath: "nextPageLink",
    mapper: {
        required: true,
        serializedName: "nextLink",
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var resourceGroupName = {
    parameterPath: "resourceGroupName",
    mapper: {
        required: true,
        serializedName: "resourceGroupName",
        type: {
            name: "String"
        }
    }
};
export var ruleName = {
    parameterPath: "ruleName",
    mapper: {
        required: true,
        serializedName: "ruleName",
        type: {
            name: "String"
        }
    }
};
export var subscriptionId = {
    parameterPath: "subscriptionId",
    mapper: {
        required: true,
        serializedName: "subscriptionId",
        type: {
            name: "String"
        }
    }
};
