// @azure/arm-rediscache/esm/operations/firewallRules.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/firewallRulesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a FirewallRules. */
var FirewallRules = /** @class */ (function () {
    /**
     * Create a FirewallRules.
     * @param {RedisManagementClientContext} client Reference to the service client.
     */
    function FirewallRules(client) {
        this.client = client;
    }
    FirewallRules.prototype.listByRedisResource = function (resourceGroupName, cacheName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            cacheName: cacheName,
            options: options
        }, listByRedisResourceOperationSpec, callback);
    };
    FirewallRules.prototype.createOrUpdate = function (resourceGroupName, cacheName, ruleName, parameters, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            cacheName: cacheName,
            ruleName: ruleName,
            parameters: parameters,
            options: options
        }, createOrUpdateOperationSpec, callback);
    };
    FirewallRules.prototype.get = function (resourceGroupName, cacheName, ruleName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            cacheName: cacheName,
            ruleName: ruleName,
            options: options
        }, getOperationSpec, callback);
    };
    FirewallRules.prototype.deleteMethod = function (resourceGroupName, cacheName, ruleName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            cacheName: cacheName,
            ruleName: ruleName,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    FirewallRules.prototype.listByRedisResourceNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByRedisResourceNextOperationSpec, callback);
    };
    return FirewallRules;
}());
export { FirewallRules };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listByRedisResourceOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{cacheName}/firewallRules",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.cacheName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RedisFirewallRuleListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var createOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{cacheName}/firewallRules/{ruleName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.cacheName,
        Parameters.ruleName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: "parameters",
        mapper: __assign(__assign({}, Mappers.RedisFirewallRuleCreateParameters), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.RedisFirewallRule
        },
        201: {
            bodyMapper: Mappers.RedisFirewallRule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{cacheName}/firewallRules/{ruleName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.cacheName,
        Parameters.ruleName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RedisFirewallRule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{cacheName}/firewallRules/{ruleName}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.cacheName,
        Parameters.ruleName,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByRedisResourceNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RedisFirewallRuleListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
