// @azure/arm-rediscache/esm/operations/patchSchedules.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import { __assign } from "tslib";
import * as msRest from "@azure/ms-rest-js";
import * as Mappers from "../models/patchSchedulesMappers";
import * as Parameters from "../models/parameters";
/** Class representing a PatchSchedules. */
var PatchSchedules = /** @class */ (function () {
    /**
     * Create a PatchSchedules.
     * @param {RedisManagementClientContext} client Reference to the service client.
     */
    function PatchSchedules(client) {
        this.client = client;
    }
    PatchSchedules.prototype.listByRedisResource = function (resourceGroupName, cacheName, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            cacheName: cacheName,
            options: options
        }, listByRedisResourceOperationSpec, callback);
    };
    PatchSchedules.prototype.createOrUpdate = function (resourceGroupName, name, scheduleEntries, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            name: name,
            scheduleEntries: scheduleEntries,
            options: options
        }, createOrUpdateOperationSpec, callback);
    };
    PatchSchedules.prototype.deleteMethod = function (resourceGroupName, name, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            name: name,
            options: options
        }, deleteMethodOperationSpec, callback);
    };
    PatchSchedules.prototype.get = function (resourceGroupName, name, options, callback) {
        return this.client.sendOperationRequest({
            resourceGroupName: resourceGroupName,
            name: name,
            options: options
        }, getOperationSpec, callback);
    };
    PatchSchedules.prototype.listByRedisResourceNext = function (nextPageLink, options, callback) {
        return this.client.sendOperationRequest({
            nextPageLink: nextPageLink,
            options: options
        }, listByRedisResourceNextOperationSpec, callback);
    };
    return PatchSchedules;
}());
export { PatchSchedules };
// Operation Specifications
var serializer = new msRest.Serializer(Mappers);
var listByRedisResourceOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{cacheName}/patchSchedules",
    urlParameters: [
        Parameters.subscriptionId,
        Parameters.resourceGroupName,
        Parameters.cacheName
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RedisPatchScheduleListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var createOrUpdateOperationSpec = {
    httpMethod: "PUT",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{name}/patchSchedules/{default}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.defaultParameter,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    requestBody: {
        parameterPath: {
            scheduleEntries: "scheduleEntries"
        },
        mapper: __assign(__assign({}, Mappers.RedisPatchSchedule), { required: true })
    },
    responses: {
        200: {
            bodyMapper: Mappers.RedisPatchSchedule
        },
        201: {
            bodyMapper: Mappers.RedisPatchSchedule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var deleteMethodOperationSpec = {
    httpMethod: "DELETE",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{name}/patchSchedules/{default}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.defaultParameter,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {},
        204: {},
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var getOperationSpec = {
    httpMethod: "GET",
    path: "subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/Redis/{name}/patchSchedules/{default}",
    urlParameters: [
        Parameters.resourceGroupName,
        Parameters.name,
        Parameters.defaultParameter,
        Parameters.subscriptionId
    ],
    queryParameters: [
        Parameters.apiVersion
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RedisPatchSchedule
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
var listByRedisResourceNextOperationSpec = {
    httpMethod: "GET",
    baseUrl: "https://management.azure.com",
    path: "{nextLink}",
    urlParameters: [
        Parameters.nextPageLink
    ],
    headerParameters: [
        Parameters.acceptLanguage
    ],
    responses: {
        200: {
            bodyMapper: Mappers.RedisPatchScheduleListResult
        },
        default: {
            bodyMapper: Mappers.CloudError
        }
    },
    serializer: serializer
};
